packages/caver-wallet/src/keyring/keyring.js

```diff
@@ -0,0 +1,845 @@
+
+        return new Keyring(address, key)
+    }
+
+    /**
+     * creates a keyring instance from address and multiple private key strings.
+     *
+     * @param {string} address An address of keyring.
+     * @param {Array.<string>} keyArray An array of private key strings.
+     * @return {Keyring}
+     */
+    static createWithMultipleKey(address, keyArray) {
+        if (!isMultipleKeysFormat(keyArray))
+            throw new Error(`Invalid format of parameter. 'keyArray' should be an array of private key strings.`)
+
+        return new Keyring(address, keyArray)
+    }
+
+    /**
+     * creates a keyring instance from address and multiple private key strings.
+     *
+     * @param {string} address An address of keyring.
+     * @param {Array.<Array.<string>>} roledBasedKeyArray A two-dimensional array containing arrays of private key strings for each role.
+     * @return {Keyring}
+     */
+    static createWithRoleBasedKey(address, roledBasedKeyArray) {
+        if (!isRoleBasedKeysFormat(roledBasedKeyArray))
+            throw new Error(
+                `Invalid format of parameter. 'roledBasedKeyArray' should be in the form of an array defined as an array for the keys to be used for each role.`
+            )
+
+        return new Keyring(address, roledBasedKeyArray)
+    }
+
+    /**
+     * encrypts a keyring and returns a keystore v4 object.
+     *
+     * @param {string|Array.<string>|Array.<string>|Keyring} key The key parameter can be an instance of Keyring, a normal private key(KlaytnWalletKey format also supported),
+     *                                                           an array of private key strings, or a two-dimensional array containing arrays of private key strings for each role,
+     * @param {string} password The password to be used for encryption. The encrypted key store can be decrypted with this password.
+     * @param {object} options The options to use when encrypt a keyring. See `keyring.encrypt` for more detail about options.
+     * @return {object}
+     */
+    static encrypt(key, password, options = {}) {
+        let keyring
+        if (_.isArray(key)) {
+            if (options.address === undefined)
+                throw new Error(`The address must be defined inside the options object to encrypt multiple keys.`)
+
+            if (isRoleBasedKeysFormat(key)) {
+                keyring = Keyring.createWithRoleBasedKey(options.address, key)
+            } else if (isMultipleKeysFormat(key)) {
+                keyring = Keyring.createWithMultipleKey(options.address, key)
+            } else {
+                throw new Error(`Invalid key format.`)
+            }
+        } else if (key instanceof Keyring) {
+            keyring = key
+        } else if (_.isString(key)) {
+            if (options.address) {
+                keyring = Keyring.createWithSingleKey(options.address, key)
```

Comment:
What if key is in a format of `KlaytnWalletKey`?

Reply:
if key is in format of KlaytnWalletKey the case will be like below

1. pass 'key' parameter in format of KlaytnWalletKey
   : In this case, 'createFromKlaytnWalletKey' function will be called in createFromPrivateKey.
2. pass 'key' parameter in format of KlaytnWalletKey and define 'address' in options
   : In this case, the error message('Invalid format of parameter. Use 'fromKlaytnWalletKey' to create Keyring from KlaytnWalletKey.') will be returned. But i think this error message is not clear when user call 'encrypt', so i will add error handling logic in here.

Reply:
I added logic for that!